src/Microsoft.ML.Transforms/NAIndicatorTransform.cs

```diff
@@ -68,322 +76,547 @@ private static VersionInfo GetVersionInfo()
+        public NAIndicatorEstimator(IHostEnvironment env, params (string input, string output)[] columns)
+            : base(Contracts.CheckRef(env, nameof(env)).Register(nameof(NAIndicatorTransform)), new NAIndicatorTransform(env, columns))
         {
-            Func<IRow, int, ValueGetter<bool>> func = ComposeGetterOne<int>;
-            return Utils.MarshalInvoke(func, Infos[iinfo].TypeSrc.RawType, input, iinfo);
+            Contracts.CheckValue(env, nameof(env));
+            _columnPairs = columns;
         }
 
         /// <summary>
-        ///  Tests if a value is NA for scalars.
+        /// Initializes a new instance of <see cref="NAIndicatorEstimator"/>
         /// </summary>
-        private ValueGetter<bool> ComposeGetterOne<T>(IRow input, int iinfo)
+        /// <param name="env">The environment to use.</param>
+        /// <param name="input">The name of the input column of the transformation.</param>
+        /// <param name="output">The name of the column produced by the transformation.</param>
+        public NAIndicatorEstimator(IHostEnvironment env, string input, string output = null)
+            : this(env, (input, output ?? input))
         {
-            var getSrc = GetSrcGetter<T>(input, iinfo);
-            var isNA = Conversions.Instance.GetIsNAPredicate<T>(input.Schema.GetColumnType(Infos[iinfo].Source));
-            T src = default(T);
-            return
-                (ref bool dst) =>
-                {
-                    getSrc(ref src);
-                    dst = isNA(ref src);
-                };
         }
 
         /// <summary>
-        /// Getter generator for vector valued inputs.
+        /// Returns the schema that would be produced by the transformation.
         /// </summary>
-        private ValueGetter<VBuffer<bool>> ComposeGetterVec(IRow input, int iinfo)
+        public override SchemaShape GetOutputSchema(SchemaShape inputSchema)
         {
-            Func<IRow, int, ValueGetter<VBuffer<bool>>> func = ComposeGetterVec<int>;
-            return Utils.MarshalInvoke(func, Infos[iinfo].TypeSrc.ItemType.RawType, input, iinfo);
+            Host.CheckValue(inputSchema, nameof(inputSchema));
+            var result = inputSchema.Columns.ToDictionary(x => x.Name);
+            foreach (var colPair in _columnPairs)
+            {
+                if (!inputSchema.TryFindColumn(colPair.input, out var col) || !Conversions.Instance.TryGetIsNAPredicate(col.ItemType, out Delegate del))
+                    throw Host.ExceptSchemaMismatch(nameof(inputSchema), "input", colPair.input);
+                var metadata = new List<SchemaShape.Column>();
+                if (col.Metadata.TryFindColumn(MetadataUtils.Kinds.SlotNames, out var slotMeta))
+                    metadata.Add(slotMeta);
+                if (col.Metadata.TryFindColumn(MetadataUtils.Kinds.IsNormalized, out var normalized))
```

Comment:
> if (col.Metadata.TryFindColumn(MetadataUtils.Kinds.IsNormalized, out var normalized)) [](start = 15, length = 86)

I think previously it got added unconditionally, and here you check source column